### PR TITLE
DOM-70026 Add data url support

### PR DESCRIPTION
**Pull request by pocheung1**
### Summary
Implements data URL support in Hephaestus to enable Hugging Face model image building without requiring internal HTTP endpoints. This simplifies the architecture by allowing Dockerfiles to be embedded directly in build requests via base64-encoded data URLs.

### Changes
- Fixed misleading error message in `pkg/buildkit/buildkit.go`
- Added data URL detection and processing in `pkg/buildkit/archive/archive.go`
- Added comprehensive test coverage in `pkg/buildkit/archive/archive_test.go`
- Updated go.mod with required dependencies

### Key Features
- Supports `data:[<mediatype>][;base64],<content>` format
- Handles base64 decoding and tar extraction
- Provides detailed logging and error handling

### Testing
- New comprehensive test suite for data URL functionality
- Tested error scenarios and edge cases
- All existing tests pass

### CODE REVIEW

File: pkg/buildkit/archive/archive.go [hunk 1]
```diff
@@ -100,6 +110,57 @@ func FetchAndExtract(ctx context.Context, log logr.Logger, url, wd string, timeo
...
+	}
+
+	mediaType := parts[0]
+	data := parts[1]
+
+	log.Info("Processing data URL", "mediaType", mediaType, "dataLength", len(data))
+
+	// Decode base64 data if specified
+	var decodedData []byte
+	if strings.Contains(mediaType, "base64") {
```
[the remark this answers is not in the document]
Comment (pocheung1) [reply pkg/buildkit/archive/archive.go:137]: Good catch!  Will change this to `==` to be more precise.
